src/common/libkvs/kvs_dir.c

```diff
@@ -160,16 +187,21 @@ kvsitr_t *kvsitr_create (kvsdir_t *dir)
 
 void kvsitr_rewind (kvsitr_t *itr)
 {
-    itr->iter = json_object_iter (itr->dirdata);
+    if (itr)
+        itr->reset = true;
 }
 
 const char *kvsitr_next (kvsitr_t *itr)
 {
     const char *name = NULL;
 
-    if (itr->iter) {
-        name = json_object_iter_key (itr->iter);
-        itr->iter = json_object_iter_next (itr->dirdata, itr->iter);
+    if (itr) {
```

Comment:
possibly a unit test for bad input here as well? although it might not be necessary coverage wise for this one.